<launch>
    <arg name="do_visualization" value="true"/>


    <param name="active_map_name" type="string" value="airsim_msbuilding"/>
    <param name="map_config_file_path" type="string" value="/home/gi/GAAS/config/maps_config_sim.yaml"/>

    <param name="lidar_topic_name" type="string" value="/velodyne_points2"/>
    <param name="ndt_result_visualization" type="bool" value="$(arg do_visualization)" />
    <param name="lidar_height_to_gps" type="double" value="0.14"/>
    <param name="ndt_downsample_size" type="double" value="0.7"/>

    <param name="use_ndt" type="bool" value="false"/>

    <!--icp params-->
    <param name="icp_transformation_epsilon" type="double" value="0.05"/>
    <param name="icp_resolution" type="double" value="10.0"/>
    <param name="icp_max_correspondence_distance" value="8.0"/>
    <param name="icp_euclidean_fitness_epsilon" value="0.5"/>

    <node name="registration_localization_node" pkg="registration_localization" type="registration_localization_node" output="screen"/>
    <group if="$(arg do_visualization)">
        <param name="do_visualization" type="bool" value="$(arg do_visualization)"/>
        <param name="map_path" type="string" value="/home/gi/Downloads/GAAS_contrib_resources/map/airsim_msbuilding/output.pcd"/>
        <node name="map_publisher_node" pkg="registration_localization" type="map_publisher" output="screen"/>
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find registration_localization)/launch/registration_localization.rviz" required="false"/>
    </group>
</launch>
